### DISCUSSION TITLE
Binary sensor device class: running

### DISCUSSION THREAD

**Discussion by ludeeus** [home-assistant/architecture#645] [Entity Models] [answered]
## Context

While adding https://github.com/home-assistant/core/pull/58120 there was not really any device class that would fit that type of entity, `connection` are close but not quite right.

## Proposal

Add a new `running` device class to binary sensors.
This can be used for services and devices (like add-ons and appliances) that have 2 operation modes (started/running) and (stopped/not running) to indicate if their operation is truthy.

Suggested implementation:
- Add a new `running` device class for binary sensors.
- When the state is `True`, the frontend will render `Running`.
- When the state is `False` the frontend will render `Not running`.
- The default icon for the `True` state is set to  `mdi:circle-slice-8` 
- The default icon for the `False` state is set to  `mdi:diameter-variant` 

![image](https://user-images.githubusercontent.com/15093472/138236613-1c3eea70-0da2-4435-847e-153b196dd0c0.png)
![image](https://user-images.githubusercontent.com/15093472/138236672-a57c54b0-9748-4666-a517-83a843218afc.png)

I'm not too sold on the icons, so if you have a better suggestion please add a comment below :+1:

**Comment by frenck** [comment]
Agreed!

**Comment by balloob** [comment]
I don't like the circle with a cross, it feels like "unable to run".

Should we use play/pause icons ?

**Reply by ludeeus** [reply]
How about play/stop in that case?
![image](https://user-images.githubusercontent.com/15093472/138346618-8b08c13d-b3ea-4601-bbd5-3f1cad8015d4.png)
![image](https://user-images.githubusercontent.com/15093472/138346648-2e290d62-1058-451a-9cf5-055bfe610985.png)

**Reply by thecode** [reply]
I think the play/stop are the best option, my unraid server also uses them and it looks good:
![image](https://user-images.githubusercontent.com/1858925/138498753-c5d025bf-884b-4444-bafd-306a9e286eaa.png)

**Comment by allenporter** [comment]
Would this device class work for integrations? The context in the PR was for add-on status, though it seems similar to whether or not an integration is healthy and running vs unhealthy and not-running.

**Reply by frenck** [reply]
For integrations, we have the available status on entities.

I think for other use cases it can be external processes (e.g., docker container running, or any other program).
But in more device perspective, this can be useful for e.g., Washing machine, the dishwasher is running or not.

**Comment by ludeeus** [comment, marked as answer]
Implementation PRs:
- https://github.com/home-assistant/core/pull/58423
- https://github.com/home-assistant/frontend/pull/10400
- https://github.com/home-assistant/developers.home-assistant/pull/1106
- https://github.com/home-assistant/home-assistant.io/pull/19963

**Comment by Mariusthvdb** [comment]
tiny suggestion, considering:

<img width="166" alt="Schermafbeelding 2021-10-25 om 23 14 53" src="https://user-images.githubusercontent.com/33354141/138772016-9138d247-b190-4e8c-8e0f-e4d5e4806000.png">

maybe change that to 'add-on is not running' to keep the terminology in sync?
